cvat-ui/src/components/annotation-page/annotation-page.tsx

```diff
@@ -17,14 +19,24 @@ interface Props {
     getJob(): void;
 }
 
-export default function AnnotationPageComponent(props: Props): JSX.Element {
+type RoutePros = RouteComponentProps<{
+    tid: string;
+    jid: string;
+}>;
+
+export default function AnnotationPageComponent(props: Props & RoutePros): JSX.Element {
     const {
         job,
         fetching,
         getJob,
+        match: {
+            params,
+        },
     } = props;
 
-    if (job === null) {
+    const jid = +params.jid;
+
+    if (job === null || (job !== undefined && job.id !== jid)) {
```

Comment:
Let's implement this logic in function ``mapStateToProps`` for ``annotations-page`` component. 
There already included router props like ``tid`` and ``jid``.
We need only add a simple condition:
```ts
return {
   ... 
   job: !job || jid === job.id ?  job : null,
}
```